app/services/openid_connect_service.rb

```diff
@@ -0,0 +1,12 @@
+class OpenidConnectService
+  cattr_accessor :private_key do
+    OpenSSL::PKey::RSA.new(
+      File.read(Rails.root.join('keys/saml.key.enc')),
+      Figaro.env.saml_passphrase
+    )
+  end
+
+  def self.identity(code)
+    Identity.where(session_uuid: code).first
+  end
```

Comment:
I would make this a method in the `OpenidConnectTokenForm` since all it's doing is calling another class.